trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/credentials/http/HttpCredentialsProvider.java

```diff
@@ -49,17 +52,18 @@ public HttpCredentialsProvider(@ForHttpCredentialsProvider HttpClient httpClient
         this.httpCredentialsProviderEndpoint = config.getEndpoint();
         ObjectMapper adjustedObjectMapper = objectMapper.registerModule(new SimpleModule().addAbstractTypeMapping(Identity.class, identityClass));
         this.jsonCodec = new JsonCodecFactory(() -> adjustedObjectMapper).jsonCodec(Credentials.class);
+        this.httpHeaders = config.getHttpHeaders();
```

Comment:
See my other comment about perhaps making the headers a map instead of an optional map https://github.com/trinodb/aws-proxy/pull/139/files#r1709133908

Either way, we should copy the headers from the config into an immutable map here

Reply:
The config returns an immutable map - https://github.com/trinodb/aws-proxy/pull/139/files#diff-572cfe02bc5685fe2645c77d0152c77b94b7f4573f9c6171b1fa68a86aed4ffbR54. So, we wouldn't be needing to convert to an immutable map again

Reply:
We still usually perform copies of maps, the config class interface only specifies it returns a Map, there's no requirement that it is immutable (though it is at the moment).

We try to make sure we copy maps and collections into immutable ones wherever they are used. Immutable collections are smart enough to do nothing if copying a collection that is already immutable ([docs](https://github.com/google/guava/wiki/ImmutableCollectionsExplained#copyof-is-smarter-than-you-think))

Can we copy this into an ImmutableMap?

Reply:
It does improve the readability on the working class. I have changed to immutable map